### PR TITLE
Rodrigom/api gateway prompt engineering test

### PR DESCRIPTION
**Pull request by snova-rodrigom**
- Updated code to support fastapi and llama3 8b
- Prompt cleaning
- Created prompts for llama3 8b
- Updated readme
- Tested using sambaverse, sambanova and fastapi endpoints

### CODE REVIEW

File: prompt_engineering/streamlit/app.py [hunk 1]
```diff
@@ -46,23 +102,26 @@ def call_sambanova_llama2_70b_api(prompt: str) -> str:
         do_sample=False,
         max_tokens_to_generate=500,
         temperature=0.0,
-        select_expert="Meta-Llama-3-8B-Instruct",
+        select_expert=model_expert,
```
Comment (snova-jorgep) [inline prompt_engineering/streamlit/app.py:105]: can  we set the model in a general sel llm model?

File: prompt_engineering/streamlit/app.py [hunk 2]
```diff
@@ -46,23 +102,26 @@ def call_sambanova_llama2_70b_api(prompt: str) -> str:
...
+        selected_model (str): selected model from Streamlit
+        api_info (str): API type
+        
     Returns:
         str: completion of the input prompt
     """
+    model_expert = _get_expert(selected_model, api_info)
+    
     #SambaNova endpoint requires these env variables. You can add more kwargs or change the value of the ones already set.    
     llm = APIGateway.load_llm(
```
Comment (snova-jorgep) [inline prompt_engineering/streamlit/app.py:126]: as above can  we set the model in a general sel llm model?

**Review by snova-jorgep — CHANGES_REQUESTED** [review]

File: prompt_engineering/config.yaml [hunk 3]
```diff
@@ -1,4 +1,8 @@
-api: "sambaverse"
+api: "sambastudio" #  set either sambastudio, sambaverse or fastapi
+
+llm:
+  max_tokens_to_generate: 500
+  temperature: 0.0
```
Comment (snova-jorgep) [inline prompt_engineering/config.yaml:5]: can we add the above api as llm sub key?

**Review by snova-jorgep — CHANGES_REQUESTED** [review]

File: prompt_engineering/config.yaml (hunk 3 above)
Comment (snova-rodrigom) [reply prompt_engineering/config.yaml]: done! didn't know this was the new standard, but I've added a couple of PRs that support this change. Thanks!

**Review by snova-jorgep — APPROVED** [review]
lgtm

**Pull request merged**